pkg/backend/net/net.go

```diff
@@ -238,17 +243,25 @@ func (b Backend) do(trace types.Trace, req *http.Request) (string, []byte, error
 // If the backend timeout is positive, Call will override the context timeout
 // with the backend timeout.
 // Call ensures that the outgoing request has a UUID set.
-func (b Backend) call(ctx context.Context, trace types.Trace, u *url.URL) (string, []byte, error) {
+func (b Backend) call(ctx context.Context, trace types.Trace, u *url.URL, request string) (string, []byte, error) {
 	ctx, cancel := b.setTimeout(ctx)
 	defer cancel()
 
-	t0 := time.Now()
+	var t *prometheus.Timer
+	if b.qHist != nil { // TODO: remove condition when capi is merged with zipper
+		t = prometheus.NewTimer(b.qHist.WithLabelValues(request))
+		t.ObserveDuration()
```

Comment:
I think we shouldn't observe duration here.

Reply:
Yep, we shouldn't.

Reply:
Pushed. Thanks for finding it!